### ISSUE TITLE
Problem executing scripts APT::Update::Post-Invoke

### ISSUE THREAD

**Issue by wangzheyuan** [readthedocs/readthedocs.org#9171]
## Details

tag 8.0.0

## Actual Result
When executing inv docker.build,  I encountered a problem.

Step 8/18 : RUN apt -y update
  ---> Running in 1a0a83674ea2
Get:1 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]
Get:2 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Get:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [109 kB]
Get:4 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [83.5 kB]
Get:5 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [90.7 kB]
Get:6 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [60.3 kB]
Get:7 http://archive.ubuntu.com/ubuntu jammy/main amd64 Packages [1792 kB]
Get:8 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [61.3 kB]
Get:9 http://archive.ubuntu.com/ubuntu jammy/multiverse amd64 Packages [266 kB]
Get:10 http://archive.ubuntu.com/ubuntu jammy/restricted amd64 Packages [164 kB]
Get:11 http://archive.ubuntu.com/ubuntu jammy/universe amd64 Packages [17.5 MB]
Get:12 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [61.7 kB]
Get:13 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [77.0 kB]
Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [96.3 kB]
Fetched 20.7 MB in 17s (1196 kB/s)
Reading package lists...
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code
ERROR: Service 'server' failed to build: The command '/bin/sh -c apt -y update' returned a non-zero code: 100

**Comment by humitos** [comment]
Hi! What version of Docker engine are you using? I found this on SO that may be useful for you: https://stackoverflow.com/questions/71941032/why-i-cannot-run-apt-update-inside-a-fresh-ubuntu22-04

**Comment by wangzheyuan** [comment]
> Hi! What version of Docker engine are you using? I found this on SO that may be useful for you: https://stackoverflow.com/questions/71941032/why-i-cannot-run-apt-update-inside-a-fresh-ubuntu22-04

The version of Docker engine is 20.10.8. I upgrade the Docker engine and the command works successfully.

**Comment by anjalirai-intel** [comment]
I installed both docker version 20.10.14 and 20.10.15, but it didn't helped

**Comment by anjalirai-intel** [comment]
intel@1e1bf9a94ad9:/home/intel/anjali/gramine$ sudo apt update
Hit:1 http://archive.ubuntu.com/ubuntu jammy InRelease
Hit:2 http://archive.ubuntu.com/ubuntu jammy-updates InRelease
Hit:3 http://security.ubuntu.com/ubuntu jammy-security InRelease
Hit:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease
Reading package lists... Done
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code
intel@1e1bf9a94ad9:/home/intel/anjali/gramine$ ^C
intel@1e1bf9a94ad9:/home/intel/anjali/gramine$ exit
exit
intel@intel-Ice-Lake-Client-Platform:~/anjali/gramine$ docker --version
Docker version 20.10.15, build fd82621

**Comment by Jamesinit** [comment]
I found a resolve way.
`apt remove command-not-found`
`apt clean`
`apt update`
`apt install command-not-found`
then ,it's work!!!

**Comment by naisy** [comment]
I also encountered this issue on an old PC.
The issue was resolved by updating Docker.

This is how to install/update Docker 19.03+.
```
# Uninstall old versions
sudo apt-get remove docker docker-engine docker.io containerd runc
```
```
sudo apt-get install \
 ca-certificates \
 curl \
 gnupg \
 lsb-release
```
```
sudo mkdir -p /etc/apt/keyrings
curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
```
```
echo \
"deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
$(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
```
```
sudo apt-get update
sudo apt-get install docker-ce docker-ce-cli containerd.io docker-compose-plugin
```

I also need nvidia-docker for gpu.
 ```
curl -s -L https://nvidia.github.io/nvidia-docker/gpgkey | \
  sudo apt-key add -
distribution=$(. /etc/os-release;echo $ID$VERSION_ID)
curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/nvidia-docker.list | \
  sudo tee /etc/apt/sources.list.d/nvidia-docker.list
sudo apt-get update
```
```
sudo apt-get install -y nvidia-container-toolkit-base
nvidia-ctk --version
sudo nvidia-ctk cdi generate --output=/etc/cdi/nvidia.yaml
sudo grep "  name:" /etc/cdi/nvidia.yaml
```
```
# wait until complete
curl https://get.docker.com | sh \
  && sudo systemctl --now enable docker
```
```
distribution=$(. /etc/os-release;echo $ID$VERSION_ID) \
      && curl -fsSL https://nvidia.github.io/libnvidia-container/gpgkey | sudo gpg --dearmor -o /usr/share/keyrings/nvidia-container-toolkit-keyring.gpg \
      && curl -s -L https://nvidia.github.io/libnvidia-container/$distribution/libnvidia-container.list | \
            sed 's#deb https://#deb [signed-by=/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg] https://#g' | \
            sudo tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
```
```
sudo rm /etc/apt/sources.list.d/nvidia-docker.list
sudo apt-get update
sudo apt-get install -y nvidia-container-toolkit
# generate /etc/docker/daemon.json
sudo nvidia-ctk runtime configure --runtime=docker
sudo systemctl restart docker
```
The problem has been resolved in this version.
```
docker -v
Docker version 23.0.5, build bc4487a
```

**Comment by mnmeyers** [comment]
I also got this issue on my mac that hadn't been updated in awhile 😅 
instructions I read most places had way too many steps so putting here what worked for me to upgrade the docker engine on my machine:
* I went to docker's [website](https://www.docker.com/products/docker-desktop/), downloaded the docker desktop version for my computer's architecture
* Quit docker desktop app
* when you drag and drop into applications, choose to replace existing version 
* (In my case I needed to upgrade my OS in order to be able to use the latest Docker desktop app)
and then you're good! Can verify with running `docker --version` or clicking the "about" menu option in the docker menu which will display the docker engine being used. 

Thanks to everyone above for your help!
